src/provider/model/Errors.ts

```diff
@@ -1,12 +1,60 @@
+export async function ErrorAggregator(items: Promise<void>[]): Promise<void> {
+    const errors: Error[] = [];
+    const promises = items.map(item => item.catch((error: Error) => errors.push(error)));
+    await Promise.all(promises);
+    if (errors.length > 0) {
+        throw new BatchError(errors);
+    }
+}
+
 abstract class CustomError extends Error {
-    public readonly type: string;
     public readonly innerError?: Error;
 
-    constructor(type: string, message: string, innerError?: Error) {
-        super(`${type}: ${message || innerError}`);
-        this.type = type;
+    constructor(name: string, message: string, innerError?: Error) {
+        super(`${message || innerError}`);
         this.innerError = innerError;
-        Object.setPrototypeOf(this, new.target.prototype);
+        this.name = name;
+    }
+}
+
+type Constructor<T> = {new(...args: any[]): T};
+
+export class BatchError extends CustomError {
```

Comment:
Would be good to discuss some of the utils in here - things like `log` seem a little out of place, and `handleError` results in error handling code that could probably be made a bit more concise.